client/src/components/relatedProducts/Modal.jsx

```diff
@@ -9,38 +10,69 @@ const Modal = (props) => {
   if (comparedProduct.product !== undefined && comparedProduct.product.features !== undefined) {
     // eslint-disable-next-line no-plusplus
     for (let i = 0; i < currentProduct.features.length; i++) {
-      if (currentProduct.features[i].value === true) {
-        currentProduct.features[i].value = <FontAwesomeIcon icon={'faCheck'} />;
-      } else if (currentProduct.features[i].value === false) {
-        currentProduct.features[i].value = '';
-      }
       features[currentProduct.features[i].feature] = {
         currentProduct: currentProduct.features[i].value,
       };
     }
     // eslint-disable-next-line no-plusplus
     for (let j = 0; j < comparedProduct.product.features.length; j++) {
-      if (comparedProduct.product.features[j].value === true) {
-        comparedProduct.product.features[j].value = <FontAwesomeIcon icon={'faCheck'} />;
-      } else if (comparedProduct.product.features[j].value === false) {
-        comparedProduct.product.features[j].value = '';
+      if (!features[comparedProduct.product.features[j].feature]) {
+        features[comparedProduct.product.features[j].feature] = {
+          comparedProduct: comparedProduct.product.features[j].value,
+        };
+      } else {
+        features[comparedProduct.product.features[j].feature].comparedProduct = (
+          comparedProduct.product.features[j].value
+        );
       }
-      features[comparedProduct.product.features[j].feature] = {
-        comparedProduct: comparedProduct.product.features[j].value,
-      };
     }
   }
 
   const keys = Object.keys(features);
   // eslint-disable-next-line react/jsx-key
-  const rows = keys.map((key) => (
-      // eslint-disable-next-line react/jsx-key
-      <tr className='features'>
-        <td className='check'>{features[key].currentProduct}</td>
-        <td>{key}</td>
-        <td className='check'>{features[key].comparedProduct}</td>
+  const rows = keys.map((key) => {
+    if (features[key].currentProduct === true && features[key].comparedProduct === true) {
+      return (
+        <tr className='features'>
+          <td className='check'>
+            <FontAwesomeIcon data-testid='checkMark' icon={ faCheck } />
```

Comment:
If the icon is added to the library then here it can be referred to as `<FontAwesomeIcon data-testid='checkMark' icon='check' />` I believe? and we wouldn't need the import on line 2. (learned from @shabbyblue16  lol)

Reply:
I can't check manually until we have either a current or related product with the appropriate value.  The tests only check for the testId. Will flag this for potential future refactoring.